numba_dppy/tests/kernel_tests/test_barrier.py

```diff
@@ -81,18 +91,22 @@ def test_local_memory(filter_str):
         pytest.skip()
     blocksize = 10
 
-    # @dppy.kernel("void(float32[::1])")
-    @dppy.kernel
-    def reverse_array(A):
-        lm = dppy.local.array(shape=10, dtype=np.float32)
-        i = dppy.get_global_id(0)
-
-        # preload
-        lm[i] = A[i]
-        # barrier local or global will both work as we only have one work group
-        dppy.barrier(dppy.CLK_LOCAL_MEM_FENCE)  # local mem fence
-        # write
-        A[i] += lm[blocksize - 1 - i]
+    try:
+
+        @dppy.kernel("void(float32[::1])")
+        def reverse_array(A):
+            lm = dppy.local.array(shape=10, dtype=np.float32)
+            i = dppy.get_global_id(0)
+
+            # preload
+            lm[i] = A[i]
+            # barrier local or global will both work as we only have one work group
+            dppy.barrier(dppy.CLK_LOCAL_MEM_FENCE)  # local mem fence
+            # write
+            A[i] += lm[blocksize - 1 - i]
+
+    except:
+        pytest.skip()
```

Comment:
It is strange that if it does not work then it is skipped. Why it could not work?

Reply:
Tests should pass or fail. They should not skip on fail. If we expect some situation incompatible with test then we should test it preliminary as a prerequisite.

Reply:
@PokhodenkoSA it could not work because `dpctl` does not support level0 yet on Windows.

Can you please suggest an example of the prerequsite? I am thinking of testing if level0 bakcend is supported when the platform is Windows and depending on that skip tests if it is not supported. But in that case we again end up skipping on fail of the prerequisite. In that case the good thing is the failure is known and in the current case it is not. Please share your thoughts.

Reply:
> Can you please suggest an example of the prerequisite?

Before running test check that Level Zero is supported.

> But in that case we again end up skipping on fail of the prerequisite.

When you test expected state of the system like checking that Level Zero is available before running test code you explicitly show to reader of the code that it is necessary to have Level Zero for the code in test to work.
Otherwise user could think that exceptions in this code is expected and normal behavior. So test does not show some stable results users can rely on.